src/thriveopedia/pages/ThriveopediaPatchMapPage.cs

```diff
@@ -104,6 +104,10 @@ protected override void Dispose(bool disposing)
     private void UpdatePlayerPatch(Patch? patch)
     {
         mapDrawer.PlayerPatch = patch ?? playerPatchOnEntry;
+
+        if (mapDrawer.Map!.UpdatePatchVisibility(mapDrawer.PlayerPatch))
```

Comment:
Please add an exception throw here if the map is null something like `throw new InvalidOperationException("This can be only called after map is set from current game");`